app/helpers/results_helper.rb

```diff
@@ -9,19 +9,42 @@ def render_field_item(doc)
         doc[:document][doc[:field]].join(", ").html_safe
   end
 
+  # truncate abstract to 450 characters
+  def render_abstract(doc)
+    doc[:document][doc[:field]][0].truncate(450).html_safe
+  end
   ##
   # Render clean faceted link to items in series
   def render_series_facet_link(doc)
-    series = doc[:document][doc[:field]]
-    collection = doc[:document][Solrizer.solr_name("collection", :displayable)].first
-    links_to_series = []
+    series = doc[:document][Solrizer.solr_name("parent_unittitles", :displayable)]
+   
+    coll = doc[:document][Solrizer.solr_name("collection", :displayable)].first
 
-    series.each do |ser|
-      links_to_series << link_to(ser, add_clean_facet_params_and_redirect([series_facet, ser],[collection_facet, collection]))
+    if doc[:document][Solrizer.solr_name("unittitle", :displayable)].nil?
+      item_title =  doc[:document][doc[:field]][0].split(">>").last.html_safe
+    else
+      item_title = doc[:document][Solrizer.solr_name("unittitle", :displayable)][0]
     end
-    [links_to_series].join(" >> ").html_safe
+  
+    title = content_tag(:span,item_title, class: "result_ut") if item_title != coll  # don't get title if it is a collection
+    coll_link = link_to(coll,add_clean_facet_params_and_redirect([collection_facet, coll]))
+    links_to_series = []
+    
+    series.each do |ser|
+      links_to_series << link_to(ser, add_clean_facet_params_and_redirect([series_facet, ser],[collection_facet, coll]))
+    end unless doc[:document][Solrizer.solr_name("parent_unittitles", :displayable)].nil? # if there is no series info at all 
+
+    [coll_link,links_to_series,title].reject(&:blank?).join(" >> ").html_safe
+    
   end
 
+  # Using this function to call series_facet_link 
+  # if parent_unittitles don't exist
+  def render_breadcrumb(doc)
```

Comment:
@eshadatta I can't find that this function is called anywhere else in the code and I don't believe it's necessary. You can delete it.